### DISCUSSION TITLE
Hyprland doesn't call .bashrc on keybinded commands

### DISCUSSION THREAD

**Discussion by ghost** [hyprwm/Hyprland#1405] [Q&A] [answered]
I have the following in my `hyprland.conf`:

```
bind = $mainMod CTRL, R, exec, bash -c "killall waybar; waybar"
```

But when I press the keybindings some custom modules are not shown.

The bar is shown correctly if I run this command from the command line.

Since I recently installed conda, my best guess currently is that it is running under a different python version, probably under a different user (maybe root user).

But I cannot fix it if I can't reproduce it in the command line.

Why in my terminal the same command has a different outcome than running it from a keybinding?

Are the two environments the same?

**Comment by ghost** [comment]
It looks like binded command are run in an environment which doesn't define `$XDG_CONFIG_HOME`.

I define this variable in my `.bashrc`.

Does hyrpland call `.bashrc` on the environment in which the keybinded commands are run?

**Comment by ghost** [comment]
It definitely looks like Hyprland changed the way the environment in which the keybinded commands are built in these days

**Reply by vaxerski** [reply]
if you can do a `git bisect` and provide the faulty commit then it would help.

but generally, yes. Hyprland calls `sh -c "WAYLAND_DISPLAY=[runtime] DISPLAY=[runtime] <your_command>"`

**Reply by ghost** [reply]
That still doesn't explain it.

`sh -c "echo $XDG_CONFIG_HOME"` shows the value of the variable I have defined in my `.bashrc`

**Reply by vaxerski** [reply]
maybe hyprland doesn't receive the envvars? how are you launching it? tty?

iirc we use the same method of launching as sway

**Reply by ghost** [reply]
I'm launching it through sddm

**Reply by vaxerski** [reply]
does it happen with tty launches too?

**Reply by ghost** [reply]
so you want me to press CTRL+ALT+F3, type hyprland and press enter, right?

**Reply by vaxerski** [reply]
si senior, or your wrapper if you use one.

**Reply by ghost** [reply]
let's continue on discord

**Comment by ghost** [comment, marked as answer]
Looked like it was a SDDM problem
